src/Pact/Compile.hs

```diff
@@ -281,10 +281,8 @@ listLiteral = withList Brackets $ \ListExp{..} -> do
   ls <- case _listList of
     _ : CommaExp : _ -> valueLevel `sepBy` sep Comma
     _                -> many valueLevel
-  let lty = case nub (map typeof ls) of
-              [Right ty] -> ty
-              _ -> TyAny
-  pure $ TList ls lty _listInfo
+  lty <- freshTyVar
+  return $ TList ls lty _listInfo
```

Comment:
My understanding was that the type in `TList` was just used for runtime typechecking. What's it used for now (other than typechecking, which could work without)?

Reply:
got it :+1: